diff --git a/node_modules/on-change/index.js b/node_modules/on-change/index.js
index 1677b09..8c73896 100644
--- a/node_modules/on-change/index.js
+++ b/node_modules/on-change/index.js
@@ -141,7 +141,9 @@ const onChange = (object, onChange, options = {}) => {
 			}
 
 			const value = isBuiltinWithMutableMethods(target)
-				? Reflect.get(target, property)
+				? target.constructor.name.includes('Date') && property !== 'constructor' && typeof property !== 'symbol'
+					? Reflect.get(target, property, receiver).bind(target)
+					: Reflect.get(target, property)
 				: Reflect.get(target, property, receiver);
 
 			return prepareValue(value, target, property);
